nan.h

```diff
@@ -136,15 +136,77 @@ namespace Nan {
 
 //=== RegistrationFunction =====================================================
 
-#if NODE_MODULE_VERSION < IOJS_3_0_MODULE_VERSION
-  typedef v8::Handle<v8::Object> ADDON_REGISTER_FUNCTION_ARGS_TYPE;
+typedef void (*addon_reg_func_t)(v8::Local<v8::Object> target);
+
+#if NODE_MODULE_VERSION < NODE_0_10_MODULE_VERSION
+namespace imp {
+  template<addon_reg_func_t F>
+  static inline void addon_reg_func(v8::Handle<v8::Object> target) {
+    v8::HandleScope scope;
+    F(v8::Local<v8::Object>::New(target));
+  }
+}
+
+# define NAN_MODULE(modname, regfunc)                                          \
```

Comment:
While we are at it, we could DRY up and drop `modname` all together. Newer versions of node-gyp define [`NODE_GYP_MODULE_NAME`](https://github.com/nodejs/node-gyp/blob/master/addon.gypi#L14). For older versions it is easily emulated since we have a `.gypi` file.

Just a thought, though.